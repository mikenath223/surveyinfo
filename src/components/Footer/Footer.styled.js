import styled from 'styled-components';
import { Row } from 'react-bootstrap';

export const FooterBar = styled(Row)`
  width: 80%;
  margin: auto;
`;

export const CopyrightText = styled.div`
  text-align: center;
  margin: 1.5rem 0;
  font-size: 15px;
  color: ${({ theme }) => theme.colors.introText};
`;

export const Divider = styled.hr`
  border-top: 1px solid grey;
  margin-top: 4rem;
  margin-bottom: 0;
  opacity: 0.3;
`;
