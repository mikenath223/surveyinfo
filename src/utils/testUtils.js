import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export const withResultThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  </ThemeProvider>)