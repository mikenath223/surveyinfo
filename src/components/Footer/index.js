import React from 'react';
import { Col } from 'react-bootstrap';
import { FooterBar, CopyrightText, Divider } from './Footer.styled';

const Footer = () => <FooterBar>
  <Col sm={12}>
    <Divider />
  </Col>
  <Col sm={12}>
    <CopyrightText>Â© 2020 CheabanTest. All rights reserved.</CopyrightText>
  </Col>
</FooterBar>;

export default Footer;