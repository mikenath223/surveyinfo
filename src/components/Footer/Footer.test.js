import React from 'react';
import 'jest-styled-components';
import Footer from './index';
import { withThemeProvider } from 'utils/testUtils';

describe('Footer', () => {
  const { container } = withThemeProvider(<Footer />);
  
  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
})
