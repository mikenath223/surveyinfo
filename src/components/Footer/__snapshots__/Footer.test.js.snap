// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<div>
  <div
    class="iIKvBE row"
  >
    <div
      class="col-sm-12"
    >
      <hr
        class="hOUiTo"
      />
    </div>
    <div
      class="col-sm-12"
    >
      <div
        class="iWfnjv"
      >
        Â© 2020 CheabanTest. All rights reserved.
      </div>
    </div>
  </div>
</div>
`;
